### PR TITLE
Add astropy_helpers to the rendered branch

### LINKED ISSUE #253: Rendered branch should have astropy-helpers
The current ``rendered`` branch example doesn't have astropy-helpers as a git submodule, that should be fixed. See discussion in #247. 

#252 is also related.

### LINKED ISSUE #258: missing ah_bootstrap.py
I used cookiecutter to get the template, but the ah_bootstrap.py seems to be missing (it's required in line 4 of the setup.py)

### PR DESCRIPTION
**Pull request by Cadair**
It's a bash script, I wrote it, it's going to break.

fixes #258 and fixes #253

### CODE REVIEW

File: .travis-deploy.sh [hunk 1]
```diff
@@ -17,5 +17,19 @@ if [[ "${TRAVIS_PULL_REQUEST}" = "false" && "$TRAVIS_OS_NAME" = "linux" && $TASK
...
+
+    # Add astropy_helpers manually at the version in the cookiecutter template
+    git submodule add https://github.com/astropy/astropy-helpers astropy_helpers || true
+    git submodule update --init
+    cd astropy_helpers
+    # debug
+    cat ../../pacakge-template/cookiecutter.json
+    # parse the json with jq to get the helpers version
+    git checkout $(jq -r ".astropy_helpers_version" ../../pacakge-template/cookiecutter.json)
+    cp ah_bootstrap.py ../
```
Comment (bsipocz) [inline .travis-deploy.sh]: you probably still need ``ez_setup.py`` here, too.

File: .travis-deploy.sh [hunk 2]
```diff
@@ -17,5 +17,19 @@ if [[ "${TRAVIS_PULL_REQUEST}" = "false" && "$TRAVIS_OS_NAME" = "linux" && $TASK
...
+    cd ..
+    git add astropy_helpers ah_bootstrap.py
+    # we might not have changes to commit
+    git commit "Update astropy_helpers" || true
```
Comment (bsipocz) [inline .travis-deploy.sh]: please mention the version number in the commit message, too.

File: .travis-deploy.sh [hunk 3]
```diff
@@ -17,5 +17,19 @@ if [[ "${TRAVIS_PULL_REQUEST}" = "false" && "$TRAVIS_OS_NAME" = "linux" && $TASK
     rsync -avz --delete --exclude .git/ ../packagename/ ./
     git add -A
     git commit -m "Update rendered version to ""$TRAVIS_COMMIT"
+
+    # Add astropy_helpers manually at the version in the cookiecutter template
+    git submodule add https://github.com/astropy/astropy-helpers astropy_helpers || true
```
Comment (bsipocz) [inline .travis-deploy.sh]: why would you need ``|| true``?

Comment (Cadair) [reply .travis-deploy.sh]: because once this is in the rendered branch the submodule will come with the clone (right?!)

File: .travis-deploy.sh [hunk 4]
```diff
@@ -25,13 +25,14 @@ if [[ "$TRAVIS_OS_NAME" = "linux" && $TASK = 'render' ]]; then
...
+    helpers_version=$(jq -r ".astropy_helpers_version" ../../cookiecutter.json)
     # parse the json with jq to get the helpers version
-    git checkout $(jq -r ".astropy_helpers_version" ../../pacakge-template/cookiecutter.json)
+    git checkout $helpers_version
     cp ah_bootstrap.py ../
     cd ..
     git add astropy_helpers ah_bootstrap.py
     # we might not have changes to commit
-    git commit "Update astropy_helpers" || true
+    git commit "Update astropy_helpers to ""$helpers_version" || true
```
Comment (bsipocz) [inline .travis-deploy.sh]: hmm, the ``-m`` is still missing. also no need for the ``""`` in the middle

**Comment by Cadair** [comment]
@bsipocz @astrofrog This fixes things can we merge it?

File: .travis-deploy.sh [hunk 5]
```diff
@@ -17,5 +20,20 @@ if [[ "${TRAVIS_PULL_REQUEST}" = "false" && "$TRAVIS_OS_NAME" = "linux" && $TASK
...
+    helpers_version=$(jq -r ".astropy_helpers_version" $base_dir/cookiecutter.json)
+    # parse the json with jq to get the helpers version
+    git checkout $helpers_version
+    cp ah_bootstrap.py ../
+    cp ez_setup.py ../
+    cd ..
+    git add astropy_helpers ah_bootstrap.py ez_setup.py
+    git status
+    # we might not have changes to commit
+    git commit -m "Update astropy_helpers to ""$helpers_version" || true
```
Comment (bsipocz) [inline .travis-deploy.sh]: why is git status needed here?

File: .travis-deploy.sh [hunk 6]
```diff
@@ -17,5 +20,20 @@ if [[ "${TRAVIS_PULL_REQUEST}" = "false" && "$TRAVIS_OS_NAME" = "linux" && $TASK
...
+    git submodule update --init
+    cd astropy_helpers
+    helpers_version=$(jq -r ".astropy_helpers_version" $base_dir/cookiecutter.json)
+    # parse the json with jq to get the helpers version
+    git checkout $helpers_version
+    cp ah_bootstrap.py ../
+    cp ez_setup.py ../
+    cd ..
+    git add astropy_helpers ah_bootstrap.py ez_setup.py
+    git status
```
Comment (bsipocz) [inline .travis-deploy.sh]: Oluwaseun Talabi

Comment (bsipocz) [inline .travis-deploy.sh]: why is git status needed here?

Comment (Cadair) [reply .travis-deploy.sh]: it's not

File: .travis-deploy.sh (hunk 5 above)
Comment (Cadair) [reply .travis-deploy.sh]: we might be running into the limits of my bash here. If `git add` leads to no changes being commited `git commit` fails.

File: .travis-deploy.sh (hunk 3 above)
Comment (bsipocz) [reply .travis-deploy.sh]: I'm not sure I follow. If astropy-helpers is already in the rendered then you don't need to init it, and you already have ah_bootstap, too, don't you? probably I have to run this script locally to see the full logic.

Comment (bsipocz) [reply .travis-deploy.sh]: oh, ok. Then just lets agree that the script needs preferably to be cleaned up after the first build, or just put all of those properly in an if statement.

Comment (Cadair) [reply .travis-deploy.sh]: You are more than welcome to improve my bash, you have many more skills with it than I do.

**Pull request merged**